htdocs/js/contextualhover.js

```diff
@@ -333,14 +333,18 @@ ContextualPopup.renderPopup = function (ctxPopupId) {
             if (!data.is_closed_membership || data.is_member) {
                 var membershipLink  = document.createElement("a");
 
-                var membership_action = data.is_member ? "leave" : "join";
-
                 if (data.is_member) {
                     membershipLink.href = data.url_leavecomm;
                     membershipLink.innerHTML = "Leave";
+                    var membership_action = "leave"
```

Comment:
I do see one possible issue here, which is that you've moved the variable scope of membership_action (the "var" declaration) into the conditional, which may mean (if the rules for JS are the same as for Perl) that the assignment is lost outside of that conditional.  I would keep the scope declaration above:

 `var membership_action;`

And then assign to the variable in each conditional without the var qualifier:

`membership_action = "leave"`
`membership_action = "accept"`
`membership_action = "join"`